zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclProperties.java

```diff
@@ -120,6 +130,19 @@ public static ZNodeGroupAclProperties getInstance() {
     return superUserDomainNames;
   }
 
+  public static boolean shouldSetX509ClientIdAsAcl() {
+    return QuorumPeerConfig.isSetX509ClientIdAsAclEnabled() && !ZNodeGroupAclProperties
+        .getInstance().isConnectionFilteringEnabled();
+  }
+
+  public boolean isConnectionFilteringEnabled() {
+    return isConnectionFilteringEnabled;
```

Comment:
This boolean is purely derived from `serverDedicatedDomain`. Do we absolutely need this?

Reply:
There are several places we need this check, I made this variable to avoid having duplicated code `if (serverDedicatedDomain != null && !serverDedicatedDomain.isEmpty())` every time.

Reply:
After making another change (make `isSetX509ClientIdAsAclEnabled` and `isConnectionFilteringEnabled` not mutually exclusive), this check will only be perform once. So I removed the `isConnectionFilteringEnabled` method.